commons/src/main/java/com/powsybl/commons/datasource/DataSourceUtil.java

```diff
@@ -63,7 +63,7 @@ static DataSource createDataSource(Path directory, String fileNameOrBaseName, Da
         Objects.requireNonNull(fileNameOrBaseName);
 
         if (fileNameOrBaseName.endsWith(".zip")) {
-            return new ZipFileDataSource(directory, getBaseName(fileNameOrBaseName.substring(0, fileNameOrBaseName.length() - 4)), observer);
+            return new ZipFileDataSource(directory, fileNameOrBaseName, getBaseName(fileNameOrBaseName.substring(0, fileNameOrBaseName.length() - 4)), observer);
         } else if (fileNameOrBaseName.endsWith(".gz")) {
             return new GzFileDataSource(directory, getBaseName(fileNameOrBaseName.substring(0, fileNameOrBaseName.length() - 3)), observer);
         } else if (fileNameOrBaseName.endsWith(".bz2")) {
```

Comment:
no, the xml importer supports network.iidm.xml as an extension so right now it must be the first dot

Reply:
The alternative is to allow the datasource to get the information from all currently loaded importers about extensions, but then we have a weird design with cyclic dependencies (not a problem from a technical point of view) where importer use datasources to check for files and datasources use importers to guess basenames (strip the importer specific extensions). Then you have to create one datasource per importer because the basename will be different depending on the importer specific extensions. It's too complex.